runtimes/azure-client-runtime/src/main/java/com/microsoft/azure/AzureClient.java

```diff
@@ -806,15 +855,23 @@ public void run() {
                         && !pollingState.getLocationHeaderLink().isEmpty()) {
                     this.serviceCall.newCall(updateStateFromLocationHeaderOnPutAsync(pollingState, pollingCallback));
                 } else {
-                    this.serviceCall.newCall(updateStateFromGetResourceOperationAsync(pollingState, url, pollingCallback));
+                    this.serviceCall.newCall(updateStateFromGetResourceOperationAsync(pollingState, url, null, pollingCallback));
                 }
             } else {
                 if (AzureAsyncOperation.SUCCESS_STATUS.equals(pollingState.getStatus()) && pollingState.getResource() == null) {
-                    this.serviceCall.newCall(updateStateFromGetResourceOperationAsync(pollingState, url, clientCallback));
+                    this.serviceCall.newCall(updateStateFromGetResourceOperationAsync(pollingState, url, serviceCall, clientCallback));
                 } else if (AzureAsyncOperation.getFailedStatuses().contains(pollingState.getStatus())) {
-                    clientCallback.failure(new ServiceException("Async operation failed"));
+                    ServiceException t = new ServiceException("Async operation failed");
```

Comment:
do you think it's useful if we include the failed status as well in this message  e.g. "Async operation failed StatusCode:500"